### PR TITLE
Update instruction encoding format

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
RPC parsed responses are not very client friendly.

#### Summary of Changes
- Update Transaction-related encoding formats: message account keys and instruction

New account-key formatting:
```json
"accountKeys": [
  {
    "pubkey": "4frSc8sFGfUsTMWeFmE2dg9NqgLXk63jTwwc1yGyR3dr",
    "signer": true,
    "writable": true
  },
  {
    "pubkey": "8z1JZ5K2b5LyovkCckdXJGNoyGCtXXCpmehKvbNxYgeB",
    "signer": true,
    "writable": true
  },
  {
    "pubkey": "TokenSVp5gheXUvJ6jGWGeCsgPKgnE3YgdGKRVCMY9o",
    "signer": false,
    "writable": false
  }
]
```

New instruction formatting:
```json
{
  "parsed": {
    "info": {
      "account": "8z1JZ5K2b5LyovkCckdXJGNoyGCtXXCpmehKvbNxYgeB",
      "mint": "6dQhs6hK9LFxKdRn1yQaw5TBhHVw4hzq4vbAcuaYYuaq",
      "owner": "8q3Md4QBT6Ucm1AorRfTXCAbVe6m57XspTGHdNJPvDK8"
    },
    "type": "initializeAccount"
  },
  "program": "spl-token",
  "programId": "TokenSVp5gheXUvJ6jGWGeCsgPKgnE3YgdGKRVCMY9o"
}
```

Toward #11357

### CODE REVIEW

**Comment by CriesofCarrots** [comment]
@jstarry how important is it that the accountKeys array not include signer/writable attributes if they are false?

**Comment by jstarry** [comment]
> @jstarry how important is it that the accountKeys array not include signer/writable attributes if they are false?

Not important, fine with me

File: transaction-status/src/lib.rs [hunk 1]
```diff
@@ -236,7 +236,7 @@ impl EncodedTransaction {
                     })
                 } else {
                     UiMessage::Parsed(UiParsedMessage {
-                        account_keys: parse_accounts(&transaction.message),
+                        account_keys: json!(parse_accounts(&transaction.message)),
```
Comment (jstarry) [inline transaction-status/src/lib.rs:239]: nit: could pass `Vec<ParsedAccount>` instead

File: transaction-status/src/lib.rs [hunk 2]
```diff
@@ -250,7 +250,7 @@ impl EncodedTransaction {
                                     instruction,
                                     &transaction.message.account_keys,
                                 ) {
-                                    UiInstruction::Parsed(parsed_instruction)
+                                    UiInstruction::Parsed(json!(parsed_instruction))
```
Comment (jstarry) [inline transaction-status/src/lib.rs:253]: nit: could pass `ParsedInstruction` instead

File: transaction-status/src/parse_instruction.rs [hunk 3]
```diff
@@ -31,6 +31,24 @@ pub enum ParseInstructionError {
 
     #[error("Program not parsable")]
     ProgramNotParsable,
+
+    #[error("Serde json error")]
```
Comment (jstarry) [inline transaction-status/src/parse_instruction.rs:35]: Is this exposed to the client? Maybe we could say "Internal error, please report"

File: transaction-status/src/parse_instruction.rs [hunk 4]
```diff
@@ -31,6 +31,24 @@ pub enum ParseInstructionError {
...
+pub struct ParsedInstruction {
+    pub program: String,
+    pub program_id: String,
+    pub parsed: Value,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct ParsedInstructionEnum {
+    pub instruction_type: String,
```
Comment (jstarry) [inline transaction-status/src/parse_instruction.rs:50]: Can we serialize this as `"type"`? `instructionType` is a little redundant

File: transaction-status/src/lib.rs (hunk 2 above)
Comment (CriesofCarrots) [reply transaction-status/src/lib.rs]: Had to add an additional enum to do this one. Let me know if you think it's not worth it.

Comment (jstarry) [reply transaction-status/src/lib.rs]: Ah, whoops! Looks fine

**Pull request merged**